src/main/java/catan/services/impl/PlayServiceImpl.java

```diff
@@ -135,6 +107,8 @@ public void buildRoad(UserBean user, String gameIdString, String edgeIdString) t
         building.setBuildingOwner(gameUserBean);
 
         edgeToBuildOn.setBuilding(building);
+        playUtil.updateCurrentCycleBuildingNumber(game);
+        playUtil.updateAvailableUserActions(game);
 
         gameDao.updateGame(game);
 
```

Comment:
I think this method should be very simple qoz now it has about 80 rows and it is difficult to keep in mind all logic when you read the code, so it should look like:

        log.debug("{} tries to build road at edge {} of game id {}", user, edgeIdString, gameIdString);

        validateUserNotEmpty(user);

        GameBean game = gameUtil.getGameById(gameIdString, ERROR_CODE_ERROR);
        validateGameStatusIsPlaying(game);
        validateActionIsAllowed(user, game, GameUserActions.BUILD_ROAD);

        EdgeBean edgeToBuildOn = getEdgeBeanToBuildOn(gameIdString, edgeIdString, game);
        validateUserCanBuildRoadOnEdge(user, edgeToBuildOn);
        buildRoadOnEdge(user, edgeToBuildOn);

        playUtil.updateCurrentCycleBuildingNumber(game);
        playUtil.updateAvailableUserActions(game);

        gameDao.updateGame(game);

        log.debug("User {} successfully built road at edge {} of game id {}", user.getUsername(), edgeIdString, gameIdString);